### PR TITLE
Make `i18n-helpers` a requirement

### PR DESCRIPTION
**Pull request by jooyunghan**
plus `gettext` now finds a PO file based on `book.language` for simpler setup.

### CODE REVIEW

**Comment by jooyunghan** [comment]
This should fix #434

File: i18n-helpers/src/bin/mdbook-gettext.rs [hunk 1]
```diff
@@ -66,19 +68,30 @@ fn translate(text: &str, catalog: &Catalog) -> String {
...
-        .ok_or_else(|| anyhow!("Missing preprocessor.gettext.po-file config value"))?
-        .as_str()
-        .ok_or_else(|| anyhow!("Expected a string for preprocessor.gettext.po-file"))?;
-    let catalog = po_file::parse(Path::new(path))
-        .map_err(|err| anyhow!("{err}"))
-        .with_context(|| format!("Could not parse {path} as PO file"))?;
+    let po_dir = cfg.get("po-dir").and_then(Value::as_str).unwrap_or("po");
+    let path = ctx.root.join(po_dir).join(format!("{language}.po"));
 
+    // no-op when PO file is missing
```
Comment (mgeisler) [inline i18n-helpers/src/bin/mdbook-gettext.rs:87]: Would it not be clearer to error out if the file doesn't exist? I'm thinking that the normal case is to have `book.language` unset, so _if_ it is set, it means that the user has actively asked for a non-standard language. This in turn makes me think that the user would like to be told about a missing PO file.

File: book.toml [hunk 2]
```diff
@@ -13,14 +13,14 @@ extra-watch-dirs = ["po"]
...
-before = ["gettext"]
+
+[preprocessor.gettext]
 
 [preprocessor.index]
 renderers = ["html"]
 
 [preprocessor.svgbob]
 renderers = ["html"]
-after = ["gettext"]
```
Comment (mgeisler) [inline book.toml:23]: Ah, are you able to remove this because the preprocessor order is determined by the order in `book.toml`? That's nice!

**Review by mgeisler — APPROVED** [review]

File: book.toml (hunk 2 above)
Comment (jooyunghan) [reply book.toml:23]: Yup, you're right.

File: i18n-helpers/src/bin/mdbook-gettext.rs (hunk 1 above)
Comment (jooyunghan) [reply i18n-helpers/src/bin/mdbook-gettext.rs:87]: It's for the English (the original book), which requires no translation but sets `book.language = en`. 

I was a bit hesitant as well for this _fallback_. Probably we can add `skip-for-languages = ["en"]` kind of setting.

Comment (jooyunghan) [reply i18n-helpers/src/bin/mdbook-gettext.rs:87]: And I was thinking additional "language specific override" like following:
```
[preprocessor.gettext]
en.skip = true
ko.po-file = "path-to-ko.po"
```

Comment (mgeisler) [reply i18n-helpers/src/bin/mdbook-gettext.rs:87]: > It's for the English (the original book), which requires no translation but sets `book.language = en`.

Ah... I had forgotten that we set the language even in that case.

> And I was thinking additional "language specific override" like following:
> 
> ```
> [preprocessor.gettext]
> en.skip = true
> ko.po-file = "path-to-ko.po"
> ```

Okay, do you have a use case for this right now?

**Pull request merged**